### PR TITLE
Status: Decouple `is_woa_site` from `at_options`

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request:

The `at_options` option was only used by WoA sites to store the slug of the plan currently active on the site. However, as we shifted towards storing purchases in the Atomic persistent data, the `at_options` option is no longer a reliable source to determine whether a site is a WoA site.

To prevent unexpected side effects, this PR removes the `at_options` check from the `is_woa_site`, so it only checks now the presence of the `wpcomsh` plugin.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

pdtR3Z-12U-p2 / D82683-code

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Apply these changes to a WoA site
- Make sure that `is_woa_site` still returns true

### CODE REVIEW

File: projects/packages/status/src/class-host.php [hunk 1]
```diff
@@ -14,8 +14,7 @@
  */
 class Host {
 	/**
-	 * Determine if this site is an WordPress.com on Atomic site or not looking first at the 'at_options' option.
-	 * As a fallback, check for presence of wpcomsh plugin to determine if a current site has undergone AT.
+	 * Determine if this site is an WordPress.com on Atomic site or not looking for presence of wpcomsh plugin.
```
Comment (anomiex) [inline projects/packages/status/src/class-host.php]:
```suggestion
	 * Determine if this site is an WordPress.com on Atomic site or not by looking for presence of the wpcomsh plugin.
```

File: projects/plugins/jetpack/functions.global.php [hunk 2]
```diff
@@ -144,8 +144,7 @@ function jetpack_get_future_removed_version( $version ) {
 }
 
 /**
- * Determine if this site is an WoA site or not looking first at the 'at_options' option.
- * As a fallback, check for presence of wpcomsh plugin to determine if a current site has undergone AT.
+ * Determine if this site is an WoA site or not looking for presence of wpcomsh plugin.
```
Comment (anomiex) [inline projects/plugins/jetpack/functions.global.php]:
```suggestion
 * Determine if this site is an WoA site or not by looking for presence of the wpcomsh plugin.
```

File: projects/packages/status/changelog/remove-is-woa-site-at-options-check [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Significance: patch
+Type: changed
+Comment: The `is_woa_site` function is now decoupled from the `at_options` option, but this is a WP.com change only
+
+
```
Comment (anomiex) [inline projects/packages/status/changelog/remove-is-woa-site-at-options-check]: There should probably still be something in the package's changelog here. Maybe something like this?
```suggestion
Update logic in `is_woa_site` function for host changes
```

Comment (mmtr) [reply projects/packages/status/changelog/remove-is-woa-site-at-options-check]: That could work, yeah. Thanks for the suggestion

**Review by coder-karen — APPROVED** [review]
Tested this out on both a non-WoA site and a WoA site to see if the `is_woa_site` outputs the correct result, and it does. 👍

**Comment by mmtr** [comment]
r249087-wpcom

**Pull request merged**